### PR TITLE
DNN-10617

### PR DESCRIPTION
**Pull request by mean2me**
It also solves DNN-10616

### CODE REVIEW

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx [hunk 1]
```diff
@@ -1,286 +1,163 @@
...
+        if (props.isHelp) {
+            if (props.commandList !== null && props.commandList.length > 0) {
+                return this.renderCommands();
             } else {
-                this.renderHelp();
+                return this.renderHelp();
             }
-            return;
         }
         if ((typeof fieldOrder === "undefined" || !fieldOrder || fieldOrder.length === 0) && fieldOrder !== null) {
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx]: Does this series of `ifs` look like something that could be split into smaller components?

Comment (tpluscode) [reply src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx]: More specifically it looks like each of the `render` methods should be its own component

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx [hunk 2]
```diff
@@ -1,286 +1,163 @@
...
-        let textLines = txt.split("\\n");
-        textLines.map((line) => {
-            let span = document.createElement("span");
-            cssSuffix = cssSuffix || "ok";
-            span.className = "dnn-prompt-" + cssSuffix;
-            span.innerText = line;
-            this.refs.cmdPromptOutput.appendChild(span);
-            this.newLine();
-        });
+    writeLine(txt, css = "dnn-prompt-ok") {
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx]: Should we really need any such `writeSomething` methods?

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx (hunk 1 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx]: Yes, it should be. Moreover both Input and Output components could be stateless functional components easily.
I propose to keep it as is for a transitional situation. Let's open a new issue for a second step refactoring in parallel with unit tests.

Comment (tpluscode) [reply src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx]: Did you propose to do this refactoring later and then just did it? 😂

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/Command.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import Parser from "html-react-parser";
+import Localization from "localization";
+import {sort} from "../../helpers";
+
+const Command = ({commandList, IsPaging}) => {
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/Command.jsx:6]: If you'd just rename this to plural which this component represents. `Commands` or `CommandList`?

**Review by tpluscode — APPROVED** [review]
I'm glad you slimmed down `Output`. It will be much easier to manage now

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx [hunk 4]
```diff
@@ -16,154 +17,163 @@ class Input extends Component {
...
-        if (cmd === "CLH" || cmd === "CLEAR-HISTORY") {
-            props.updateHistory("", true);
-            actions.runLocalCommand(cmd, Localization.get("SessionHisotryCleared"));
-            return;
-        }
-        if (cmd === "CONFIG") {
-            this.configConsole(tokens);
-            return;
+        let done = true;
+        switch (cmd) {
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx:63]: Much cleaner with a `switch` :+1:

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx [hunk 5]
```diff
@@ -16,154 +17,163 @@ class Input extends Component {
...
-                    }
-                } else {
-                    // found but no value
-                    return null;
-                }
-            }
-        }
-        // not found
-        return null;
+        if (!tokens || tokens.length === 0) return null;
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx]: I'd rather we keep mandatory braces. Isn't it part of our eslint setup?

Actually, is prompt set up with eslint?

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx [hunk 6]
```diff
@@ -16,154 +17,163 @@ class Input extends Component {
...
-            mode = tokens[1];
+        actions.runLocalCommand("INFO", "\n" + txt + "\n", "cmd");
+    }
+
+    runCmd() {
+        const txt = this.readInput();
+        if(txt) {
+            this.displayCmdInfo(txt);
+            if(!this.runLocalCmd(txt))
+                this.runServerCmd(txt);
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx]: Please add braces.

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx [hunk 7]
```diff
@@ -16,154 +17,163 @@ class Input extends Component {
...
         }
-        if (mode) {
-            actions.changeUserMode({ UserMode: mode.toUpperCase() }, () => {
-                if (mode.toUpperCase() === "EDIT")
+    }
+
+    changeUserMode(tokens) {
+        const { props } = this;
+        const { actions } = props;
+        if (!tokens && tokens.length >= 2) return;
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx]: Please add braces.

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx (hunk 5 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx]: Yes, we have eslint.

File: src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx (hunk 4 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx:63]: 👍

**Review by tpluscode — APPROVED** [review]

**Pull request merged**